src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/ChangeManager.java

```diff
@@ -0,0 +1,178 @@
+package uk.gov.companieshouse.overseasentitiesapi.utils;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * A utility class to compare and apply changes to a Psc object
+ *
+ * @param <T> The type that contains the data to be changed
+ */
+public class ChangeManager<T, L, R> {
+
+  private final T objectToAddChanges;
+  private final L leftPairData;
+  private final R rightPairData;
+
+
+  /**
+   * @param objectToAddChanges The object containing the data that has been changed
+   */
+  public ChangeManager(T objectToAddChanges, Pair<L, R> pair) {
+    this.objectToAddChanges = objectToAddChanges;
+    this.leftPairData = pair.getLeft();
+    this.rightPairData = pair.getRight();
+  }
+
+  private <P, C, S, U> boolean compareAndBuildChange(
+      P proposedData,
+      Function<U, C> currentDataGetter,
+      U leftOrRightDataFromPair,
+      Function<P, S> proposedConverter,
+      BiPredicate<P, C> equalityPredicate,
+      BiConsumer<T, S> dataSetter) {
+
+    var convertedProposedData = proposedConverter.apply(proposedData);
+
+    if(leftOrRightDataFromPair == null && objectToAddChanges == null){
+      return false;
+    }
+
+    if (
+        !isEmpty(convertedProposedData) &&
+        (leftOrRightDataFromPair == null || !equalityPredicate.test(proposedData, currentDataGetter.apply(leftOrRightDataFromPair)))
+    ) {
+      dataSetter.accept(objectToAddChanges, convertedProposedData);
+      return true;
+    }
+    return false;
+  }
+
+
+  /**
+   * @param proposedData      The data that could be added to objectToAddChanges
+   * @param currentDataGetter A getter from the class of the current data
+   * @param proposedConverter A function to convert the proposed data to the setter type. If they
+   *                          are the same datatype, use Function.identity()
+   * @param equalityPredicate A function to compare the proposed and current data. If they are the
+   *                          same datatype, use Objects::equals
+   * @param dataSetter        A function to set the data on the Psc object
+   * @param <P>               The type of the proposed data
+   * @param <C>               The type of the current data
+   * @param <S>               The type of the setter data
+   * @return true if the data has changed, false otherwise
+   */
+  public <P, C, S> boolean compareAndBuildLeftChange(
```

Comment:
Just to be consistent with the private counterpart method? (or change private data method to '....RightChange')
```suggestion
  public <P, C, S> boolean compareAndBuildPublicChange(
```